types/tests/augmentation-test.ts

```diff
@@ -0,0 +1,11 @@
+import Banana from '../index';
+
+declare module '../banana-i18n' {
+	type A = 'messageKey';
```

Comment:
'messageKey' as type for messageKey? I have not used augmentation, but as per my understanding, we are adding a new type for message keys other than the already declared parameter types right?

What I expect here is a real example of augmentation. Reusing the variable names for different purposes is not useful there.

Can we have a realistic example here please?

Reply:
At least we a using a generic type, which will be declared in a concrete way in the using projects, if they want to.  Practically spoken the projects have to declare (a) new type(s), if they want to use it.

A real example is quite extensive...or at least what I understand under it. I also would not see as part of the 'test' and more in the domain of sample project, where it is more obvious what people gaining from using it.

Maybe you have some more suitable in mind then me, so could maybe explain, what you understand in this case under a realistic example, please?

Reply:
Just wanna ask if you still follow this one?

Reply:
Augmenting types is a possible requirement for a project which want to use their own types. But with in the context of this project(banana-i18n),
a) Why this commit - adding types - require this piece of code? Why can't we simplify this commit and avoid code for a usecase we can't imagine in clear way? Isn't it a good practice to have simple commits?
b) Why can't the project that want to augment banana-i18n do that augmentation?

Reply:
> a) Why this commit - adding types - require this piece of code? Why can't we simplify this commit and avoid code for a usecase we can't imagine in clear way? Isn't it a good practice to have simple commits?

I already saw this use case and I can even now imagine other cases. (Unfortunately the team decided not to use banana.) Well, you are right, if you say that patches should be simple as possible, but let's think about this from another perspective: Let's say (and I think you already said somewhere) test's also are some sort of manual, doc or examples of usage. Augmentation is something which can be very helpful, if you want to define very clear / tightly typing in order to make your life easy (with a little help from your IDE or vim or at least the compiler, which can use those definitions). I remember that I was very thankful, if I found something about "how do I augment that thing" and mostly I looked up first for it in the tests. That spared my brain cells, which I could invest in other things and saved me to go deep into a rabbit hole, just to figure out augmentation is not possible or how. I think you get my point here.

> b) Why can't the project that want to augment banana-i18n do that augmentation?

You are have pointing on something, which I overlooked - indeed there is something, which is redundant, but in the type definition.